docarray/array/mixins/io/binary.py

```diff
@@ -191,18 +208,33 @@ def save_binary(
     ) -> None:
         """Save array elements into a binary file.
 
+        :param file: File or filename to which the data is saved.
+        :param protocol: protocol to use
+        :param compress: compress algorithm to use
+
+         .. note::
+            If `file` is `str` it can specify `protocol` and `compress` as file extensions.
+            This functionality assumes `file=file_name.$protocol.$compress` where `$protocol` and `$compress` refer to a
+            string interpolation of the respective `protocol` and `compress` methods.
+            For example if `file=my_docarray.protobuf.lz4` then the binary data will be created using `protocol=protobuf`
+            and `compress=lz4`.
+
         Comparing to :meth:`save_json`, it is faster and the file is smaller, but not human-readable.
 
         .. note::
             To get a binary presentation in memory, use ``bytes(...)``.
 
-        :param protocol: protocol to use
-        :param compress: compress algorithm to use
-        :param file: File or filename to which the data is saved.
         """
         if isinstance(file, io.BufferedWriter):
             file_ctx = nullcontext(file)
         else:
+            _protocol, _compress = protocol_and_compress_from_file_path(file)
+
+            if _protocol is not None:
+                protocol = _protocol
+            if _compress is not None:
+                compress = _compress
```

Comment:
I think we should enforce the presence of both, protocol and compress, in order to use any of them, as documented.

Reply:
you can decide to use No compression